lib/spack/spack/cmd/dev_build.py

```diff
@@ -45,6 +45,15 @@ def setup_parser(subparser):
     cd_group = subparser.add_mutually_exclusive_group()
     arguments.add_common_arguments(cd_group, ['clean', 'dirty'])
 
+    testing = subparser.add_mutually_exclusive_group()
```

Comment:
No need for this mutually exclusive group here. This is only present in `spack install` because of the deprecated `--run-tests` flag that conflicts with this flag.